src/main/scala/com/algorithmia/handler/ResponseHandler.scala

```diff
@@ -0,0 +1,39 @@
+package com.algorithmia.handler
+
+import java.io.{FileOutputStream, PrintStream}
+
+import play.api.libs.json.{Json, Writes}
+
+import scala.reflect.ClassTag
+import scala.util.Try
+
+case class ResponseHandler[O]() {
+  val FIFOPATH = "/tmp/algoout"
+
+  private def write(data: String): Try[Unit] = {
+    Try(new PrintStream(new FileOutputStream(this.FIFOPATH, true)))
+      .map({ s =>
+        s.println(data); s
+      })
+      .map({ s =>
+        s.flush(); s
+      })
+      .map(_.close())
+  }
```

Comment:
these should not be 3 map statements. It should probably just be one:
```
    Try(new PrintStream(new FileOutputStream(this.FIFOPATH, true)))
      .map(s => {
        s.println(data)
        s.flush
        s.close
    })
```

Reply:
accepted, agreed this is cleaner.